src/log4stash/Bulk/LogBulkSet.cs

```diff
@@ -1,25 +1,34 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using System.Linq;
-using System.Text;
-using System.Threading;
-using System.Threading.Tasks;
 using log4stash.Configuration;
 using log4stash.SmartFormatters;
 
 namespace log4stash.Bulk
 {
     public class LogBulkSet : ILogBulkSet
     {
-        private List<InnerBulkOperation> _bulk = new List<InnerBulkOperation>();
+        private List<List<InnerBulkOperation>> _allBulks = new List<List<InnerBulkOperation>>();
```

Comment:
I think you can just save the bulk sizes here and not reference the whole bulk.

Basically - save only the "currentBulk" as we do today, and a `totalSize`.
At ResetBulk() you do `totalSize += _bulk.Count`
At CommitBulk() you need to get the size and do `totalSize -= size`